php_phongo.c

```diff
@@ -507,7 +518,10 @@ zend_bool phongo_writeerror_init(zval *return_value, bson_t *bson TSRMLS_DC) /*
 		writeerror->code = bson_iter_int32(&iter);
 	}
 	if (bson_iter_init_find(&iter, bson, "errmsg") && BSON_ITER_HOLDS_UTF8(&iter)) {
-		writeerror->message = bson_iter_dup_utf8(&iter, NULL);
+		uint32_t errmsg_len;
+		const char *err_msg = bson_iter_utf8(&iter, &errmsg_len);
+
+		writeerror->message = estrndup(err_msg, errmsg_len);
```

Comment:
The bug was using inconsistent `strdup()` and `free()` functions, and the alternative fix was to use `bson_free()` in the class dtors. For consistency with other class structs, where the strings are always allocated by PHP, I chose to modify the allocation to match and leave the `efree()` in place.

Of course, the issue only presented itself when I disabled the custom BSON memory allocators, but I was uncomfortable with having this bit of unrelated code depend on that implementation detail.